mac/README.md

```diff
@@ -8,22 +8,53 @@ Install cocoapods (sudo gem install cocoapods) if not already installed.
 ## Keyman for macOS Development
 Keyman for macOS can be built from a command line (preferred) or Xcode.
 
+### Setting up your code signing and notarization environment.
+With macOS 10.14 and later, Keyman must be signed then notarized by Apple, even for local
+test builds. This requires additional configuration for your build environment.
+
+1. First, open XCode, Preferences, Accounts, and select Manage Certificates for the identity
+   you wish to use for signing. Click **+** and select **Developer ID Application**. A
+   certificate will then be generated and listed in your Keychain. Get info for the certificate
+   you generated from Keychain and copy its SHA-1 fingerprint. You'll need to remove spaces
+   from between the hex pairs; you'll use this in step 3.
+
+2. Determine the Apple ID details in order to run a build. You'll need to create an
+   App-Specific Password at https://appleid.apple.com/ and use this. Your Shortname will
+   be either your 10-digit Team Identifier or a shortname that can be extracted with
+   the following command:
+
+   `/Applications/Xcode.app/Contents/Applications/Application\ Loader.app/Contents/itms/bin/iTMSTransporter -m provider -u '<Username>' -p '<Password>' -account_type itunes_connect -v off`
+
+   Note: Use your Apple ID for `<Username>` and the app-specific password you generated above
+   for `<Password>`. You'll use these in the next step as well.
+
+3. Add the following environment variables, probably to your .bashrc file, replacing with the
+   values you collected in the previous steps:
+
+        export CERTIFICATE_ID=<SHA1-Fingerprint>
+        export APPSTORECONNECT_PROVIDER=<Shortname>
+        export APPSTORECONNECT_USERNAME=<Username>
+        export APPSTORECONNECT_PASSWORD=<Password>
+
 ### Compiling from Command Line
 To build Keyman for macOS, do the following:
 1. Open a Terminal window.
 2. cd to **keyman/mac**. **build.sh** must be run in the directory containing the script.
 3. Build using `./build.sh -no-codesign`. Run `./build.sh -help` to see all options.
-    * If you have signing credentials from the core development team, you can build a signed version by omitting
-  `-no-codesign`.
-  
+    * If you have signing credentials from the core development team, you can build a signed version by omitting `-no-codesign`. Somewhat misleadingly, `-no-codesign` only stops
+    automatic signing using the certificate maintained by the core development team!
+
 Note: If Carthage prompts you to allow it access to your github credentials, it's fine to click Deny.
 
 ### Running Keyman
 1. Deploy Keyman locally using `./build.sh -deploy local -deploy-only`.
-    * Alternatively copy **keyman/mac/Keyman4MacIM/build/Debug/Keyman.app** to **~/Library/Input Methods**
+    * This will notarize the app, signing with your local credentials if not already signed, and copy **keyman/mac/Keyman4MacIM/build/Debug/Keyman.app** to **~/Library/Input Methods**
```

Comment:
Note that this requires a "release" build, which step 3 doesn't generate... at least without the `-no-codesign` parameter.  That said, since "debug" builds seem to be useless now, maybe "release" should now be the default?

Reply:
Release should be the default for the command line build I guess. Debug builds are useful for local debugging without being installed as an IM, though.